### PR TITLE
Blurple buttons foreva - part 1

### PR DESCRIPTION
**Pull request by weerd**
### What's this PR do?

This pull request begins the process of updating buttons throughout all of Phoenix to switch to the new branding and ideally make using buttons in components all-around easier 🕺 This first part establishes 4 new button related components:

- `ButtonLink` (a button that is an `<a>` anchor link with an `href`; aiming to eventually replace the existing `ButtonLink` component).
- `ButtonElement` (a button that is a `<button>` element; aiming to eventually replace the existing `Button` component).
- `PrimaryButton` (a styled version based on new branding of a primary button and ultimately calls on a `ButtonLink` or `ButtonElement` but with specified props).
- `SecondaryButton` (a styled version based on new branding of a secondary button and ultimately calls on a `ButtonLink` or `ButtonElement` but with specified props).

Examples:

Buttons Galore

### How should this be reviewed?

👀 

### Any background context you want to provide?

In `base.scss` I commented out all the changes that are planned for the composed `.btn` class, because I'm waiting to implement those in Part 2. To reduce the size of this PR it made sense to not make that change yet, and then have to update a slew of buttons throughout the codebase.

### Relevant tickets

References [Pivotal #172184401](https://www.pivotaltracker.com/story/show/172184401).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.

### CODE REVIEW

**Comment by lindsayrmaher** [comment]
OOooOoOOO i'm so excited about this

**Comment by weerd** [comment]
> OOooOoOOO i'm so excited about this

Oh hello! 👋 😄

File: resources/assets/components/utilities/Button/ButtonElement.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/* eslint-disable react/button-has-type */
```
Comment (weerd) [inline resources/assets/components/utilities/Button/ButtonElement.js:1]: I was getting the following error:

```
error  `type` is an invalid value for button type attribute  react/button-has-type
```
It works fine, but looks like the ESLint rule is a little wonky, and doesn't like the attribute value passed as a variable 🤷‍♂️

**Comment by weerd** [comment]
One thing to note, that in upcoming PR, as I replace existing buttons, I'll probably run into hiccups, where I'll need to update these new button components, but for the most part I tested what I could for this initial setp!

**Comment by weerd** [comment]
Here is an excerpt of the JSX markup for the sample buttons in the GIF above if interested:

```jsx
<ButtonElement
  className="bg-green-500 hover:bg-green-700 mb-12"
  onClick={() => console.log('😈')}
  text="Button Element"
/>

<ButtonElement
  className="bg-green-500 hover:bg-green-700 ml-10 mb-12"
  isDisabled
  onClick={() => console.log('😈')}
  text="Button Element Disabled"
/>

<br />

<ButtonLink
  className="bg-red-500 hover:bg-red-700 mb-12"
  href="/"
  text="Button Link"
/>

<br />

<PrimaryButton
  className="mb-12"
  text="Primary Button Element"
  onClick={() => console.log('😈')}
/>

<PrimaryButton
  className="ml-10 mb-12"
  isDisabled
  text="Primary Button Element Disabled"
  onClick={() => console.log('😈')}
/>

<br />

<PrimaryButton
  className="mb-12"
  href="/"
  text="Primary Button Link"
/>

<br />

<SecondaryButton
  className="mb-12"
  onClick={() => console.log('😈')}
  text="Secondary Button Element"
/>

<SecondaryButton
  className="mb-12 ml-12"
  isDisabled
  onClick={() => console.log('😈')}
  text="Secondary Button Element Disabled"
/>

<br />

<SecondaryButton
  className="mb-12"
  href="/"
  text="Secondary Button Link"
/>
```

**Comment by weerd** [comment]
Yep! Part 2 (and maybe 3) will be converting all the buttons and determining how these components need to be modified to address nuances with how buttons are used throughout Phoenix!

**Pull request merged**